src/Core/Models/Table/EmergencyAccess.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Bit.Core.Enums;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.Models.Table
+{
+    public class EmergencyAccess : ITableObject<Guid>
+    {
+        public Guid Id { get; set; }
+        public Guid GrantorId { get; set; }
+        public Guid? GranteeId { get; set; }
+        public string Email { get; set; }
+        public string KeyEncrypted { get; set; }
+        public EmergencyAccessType Type { get; set; }
+        public EmergencyAccessStatusType Status { get; set; }
+        public int WaitTimeDays { get; set; }
+        public DateTime? RecoveryInitiatedAt { get; internal set; }
+        public DateTime? LastNotificationAt { get; internal set; }
+        public DateTime CreationDate { get; internal set; } = DateTime.UtcNow;
+        public DateTime RevisionDate { get; internal set; } = DateTime.UtcNow;
+
+        public void SetNewId()
+        {
+            Id = CoreHelpers.GenerateComb();
+        }
+        
+        public EmergencyAccess ToEmergencyAccess()
+        {
+            return new EmergencyAccess
+            {
+                Id = Id,
+                GrantorId = GrantorId,
+                GranteeId = GranteeId,
+                Email = Email,
+                KeyEncrypted = KeyEncrypted,
+                Type = Type,
+                Status = Status,
+                WaitTimeDays = WaitTimeDays,
+                RecoveryInitiatedAt = RecoveryInitiatedAt,
+                LastNotificationAt = LastNotificationAt,
+                CreationDate = CreationDate,
+                RevisionDate = RevisionDate,
+            };
+        }
```

Comment:
Am a little confused by this? Is this for copying?

Reply:
It's for converting between `EmergencyAccessDetails` to `EmergencyAccess` which is used in the notification/grant jobs. It loads the data from the view to get access to the grantor/grantee details required to send the emails, and to avoid doing another database query it copies the view result.